pytest/lib/peer.py

```diff
@@ -29,17 +30,28 @@ async def send_raw(self, raw_message):
         self.writer.write(raw_message)
         await self.writer.drain()
 
+    # returns None on timeout
     async def recv(self, expected=None):
         while True:
             response_raw = await self.recv_raw()
+            if response_raw is None:
+                return None
+
             response = BinarySerializer(schema).deserialize(
                 response_raw, PeerMessage)
 
             if expected is None or response.enum == expected:
                 return response
 
     async def recv_raw(self):
-        length = await self.reader.read(4)
+        try:
+            length = await asyncio.wait_for(self.reader.read(4), 5)
+        except concurrent.futures.TimeoutError:
+            return None
+
+        if len(length) < 4:
+            return None
```

Comment:
I see why this is necessary, but when we hit this path all the communication will be a total mess since messages won't be aligned anymore. Moreover, if at some point `len(length)` is `0` it means the connection was dropped in the other  end (which is also not handled). This is a note for me to fix this.

Reply:
The exact situation I'm trying to handle here is actually the connection dropped.